src/modules/select/wallets/cobovault.ts

```diff
@@ -0,0 +1,261 @@
+
+  let addressList = Array.from<string>([])
+  let enabled = false
+  let customPath = false
+
+  const provider = createProvider({
+    getAccounts: (callback: any) => {
+      getAccounts()
+        .then((res: Array<string | undefined>) => callback(null, res))
+        .catch((err: any) => callback(err, null))
+    },
+    signTransaction: (transactionData: any, callback: any) => {
+      signTransaction(transactionData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    rpcUrl
+  })
+
+  provider.setPath = setPath
+  provider.dPath = dPath
+  provider.enable = enable
+  provider.setPrimaryAccount = setPrimaryAccount
+  provider.getPrimaryAddress = getPrimaryAddress
+  provider.getAccounts = getAccounts
+  provider.getMoreAccounts = getMoreAccounts
+  provider.getBalance = getBalance
+  provider.getBalances = getBalances
+  provider.send = provider.sendAsync
+  provider.disconnect = disconnect
+  provider.isCustomPath = isCustomPath
+
+  function disconnect() {
+    dPath = ''
+    enabled = false
+    provider.stop()
+  }
+
+  async function setPath(path: string) {
+    if (path !== BASE_PATH)
+      throw new Error("CoboVault only supports standard path: m/44'/0'/0'/0/x")
+    customPath = false
+    dPath = path
+    return true
+  }
+
+  function isCustomPath() {
+    return customPath
+  }
+
+  function enable() {
+    enabled = true
+    return keyring.readKeyring().then(() => {
+      return getAccounts()
+    })
+  }
+
+  function addresses() {
+    return addressList
+  }
+
+  function setPrimaryAccount(address: string) {
+    return keyring.setCurrentAccount(
+      addressList.findIndex(addr => addr === address) || 0
+    )
+  }
+
+  function getPrimaryAddress() {
+    return keyring.getCurrentAddress()
+  }
+
+  async function getMoreAccounts() {
+    const accounts = await getAccounts(true)
+    return accounts && getBalances(accounts)
+  }
+
+  async function getAccounts(getMore?: boolean) {
+    if (!enabled) {
+      return []
+    }
+
+    if (addressList.length > 0 && !getMore) {
+      return addressList
+    }
+
+    try {
+      addressList = await keyring.addAccounts(5)
```

Comment:
```suggestion
      addressList = await keyring.addAccounts(addressList.length + 5)
```

What do you think about changing the above code so that when the user clicks the "load more" button on the accounts modal, it loads another 5 addresses just like our other hardware wallet modules do?

Reply:
Thanks.